### ISSUE TITLE
Long String is too Slow

### ISSUE THREAD

**Issue by IceMan0224** [mike-ward/VSColorOutput64#10]
I apologize for my poor English.
First of all, thank you for creating such a wonderful extension

The bug I found is that when outputting a very long string (e.g. cl command), there is a freeze for a while and then the output window works. 

I think regex is slow. 

Please check this issue, thank you

**Comment by pborsutzki** [comment]
I have a similar issue in my project. A legacy part of it is using build commands that are around 5500 characters long (cmake invokes cygwin, each time the whole environment is passed, which makes it rather long, and it of course it hard to change). When the plugin is active while building that legacy part, Visual Studio becomes unusable until all 150 legacy files are built. Afterwards scrolling through the output window over the long lines results in similar lag. Disabling the extension fixes the issue.

I tracked the problem down to the evaluation of the regular expressions. A `RegexMatchTimeoutException` is thrown every time [here](https://github.com/mike-ward/VSColorOutput64/blob/59424fb9bfceea04a74d9fc1995aee7673d4fd86/VSColorOutput/Output/ColorClassifier/OutputClassifier.cs#L107) while matching against the second regex of the default list (`.*[t|c]sc\.exe.*`). Interestingly, matching the same regex in Notepad++ against _the whole build output_ containing 150 of such lines is quite fast (it takes far less than a second), assuming it is only matched against single lines (which I think the plugin does). So imho matching this regex shouldn't take anywhere near the 250ms budget for 5500 chars.

First I tried caching and compiling the Regex objects [here](https://github.com/mike-ward/VSColorOutput64/blob/59424fb9bfceea04a74d9fc1995aee7673d4fd86/VSColorOutput/Output/ColorClassifier/RegExClassification.cs#L36-L44), but wasn't successful. Even when compiled and cached, the regex still runs into a timeout. As an intermediate workaround I hackfixed it by changing the timeouts [here](https://github.com/mike-ward/VSColorOutput64/blob/59424fb9bfceea04a74d9fc1995aee7673d4fd86/VSColorOutput/Output/ColorClassifier/RegExClassification.cs#L36-L44) to 25ms (currently: 250ms), which made it somewhat usable for me.

Finally, I tried modifying the problematic regex. I changed it from `.*[t|c]sc\.exe.*` to `[t|c]sc\.exe` and to my surprise it made it fast! So now my question is, do you need the `.*` in that regex? I am not sure what kind of output it is supposed to match, but I also can't find a good reason why they are there as it should just do the same without these. As far as I can see, the tests run fine without them. Of course I also don't understand why this regex implementation is so slow when the `.*` are present.

TL;DR:
-> **I suggest to change the default regex from `.*[t|c]sc\.exe.*` to `[t|c]sc\.exe`**

I also suggest to change the time budget for each regex to something a lot lower, or at least make it configurable. In case you hit the max currently, it really makes everything unusable (and even slows down the build a lot).

**Comment by pborsutzki** [comment]
I opened pull request #25 for just changing the default regex. I am not very familiar with C# development so I don't know what I would break when I'd try to add a configurable timeout to the settings. As just lowering the default timeout might break some peoples usecases, I refrained from that as well for now.
